### PR TITLE
API: Delete Token

### LINKED ISSUE #92: Add a new API for deleting a token
### Problem

This feature is not currently supported on Hedera Wallet Snap

### Solution

Deleting a token marks a token as deleted, though it will remain in the ledger. The operation must be signed by the specified Admin Key of the Token. If the Admin Key is not set, the Transaction will result in TOKEN_IS_IMMUTABlE. Once deleted update, mint, burn, wipe, freeze, unfreeze, grant KYC, revoke KYC and token transfer transactions will resolve to TOKEN_WAS_DELETED.

You cannot delete a specific NFT. You can delete the class of the NFT specified by the token ID after you have burned all associated NFTs associated with the token class

Refer to: https://docs.hedera.com/hedera/sdks-and-apis/sdks/readme-1/delete-a-token

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by poi-son-ivy**
**Description**:
This change adds a 'hts/deleteToken' API which allows a user to delete a token they have created. 

**Related issue(s)**:

Fixes #92 

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/hedera-wallet-snap/packages/site/src/components/cards/hts/DeleteToken.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
+/*-
+ *
+ * Hedera Wallet Snap
+ *
+ * Copyright (C) 2024 Tuum Tech
```
Comment (kpachhai) [inline packages/hedera-wallet-snap/packages/site/src/components/cards/hts/DeleteToken.tsx:5]: Could you change this to say Hedera Hashgraph like all the other license headers?

File: packages/hedera-wallet-snap/packages/snap/src/utils/HederaUtils.ts [hunk 2]
```diff
@@ -1480,6 +1481,35 @@ export class HederaUtils {
...
+        'Invalid deleteToken Params passed. "tokenId" must be passed as a parameter',
+      );
+      throw providerErrors.unsupportedMethod(
+        'Invalid deleteToken Params passed. "tokenId" must be passed as a parameter',
+      );
+    }
+
+    const parameter = params as DeleteTokenRequestParams;
+
+    if (_.isEmpty(parameter.tokenId) || typeof parameter.tokenId !== 'string') {
```
Comment (kpachhai) [inline packages/hedera-wallet-snap/packages/snap/src/utils/HederaUtils.ts:1503]: Can we not use checkValidString function here?

File: packages/hedera-wallet-snap/packages/site/src/components/cards/hts/DeleteToken.tsx (hunk 1 above)
Comment (poi-son-ivy) [reply packages/hedera-wallet-snap/packages/site/src/components/cards/hts/DeleteToken.tsx:5]: can do!

File: packages/hedera-wallet-snap/packages/snap/src/utils/HederaUtils.ts (hunk 2 above)
Comment (poi-son-ivy) [reply packages/hedera-wallet-snap/packages/snap/src/utils/HederaUtils.ts:1503]: we sure can, changing it!

**Comment by kpachhai** [comment]
While testing this, I get an error from Metamask:
```
VM5685:1 MetaMask - RPC Error: Error while trying to delete token 0.0.3590748 from the account: TypeError: Cannot read properties of undefined (reading 'accountId') 
{code: 4200, message: "Error while trying to delete token 0.0.3590748 fro…ead properties of undefined (reading 'accountId')"}``` @poi-son-ivy

**Review by kpachhai — APPROVED** [review]

**Pull request merged**